Tasks/Common/webdeployment-common/msdeployutility.ts

```diff
@@ -156,6 +157,41 @@ function getMSDeployLatestRegKey(registryKey: string): Q.Promise<string> {
     return defer.promise;
 }
 
+function useRequiredDotNetVersion(registryKey: string): Q.Promise<string> {
+    var defer = Q.defer<string>();
+    var regKey = new winreg({
+      hive: winreg.HKLM,
+      key:  registryKey
+    })
+
+    regKey.keys(function(err, subRegKeys) {
+        if(err) {
+            defer.reject(tl.loc("UnabletofindthelocationofDotNetFrameworkfromregistryonmachineError", err));
```

Comment:
Can you cross check with https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed

Reply:
We are in sync with this. For entire version number they go one level deeper and read the value of "version" subkey, but we don't need it.